### PR TITLE
feat: #35: rely on build artifact for publishing pr

### PR DESCRIPTION
**Pull request by matthieu-crouzet**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -36,6 +36,13 @@ jobs:
           AZ_AR_PASSWORD: ${{ secrets.AZ_AR_PASSWORD }}
       - name: Build
         run: yarn build
+      - name: Store dist folders
+        uses: actions/upload-artifact@v3
+        if: github.event_name == 'pull_request'
+        with:
+          path: |
```
Comment (kpanot) [inline .github/workflows/main.yml:43]: missing `apps` folder

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -60,7 +60,14 @@ jobs:
         run: yarn build:swagger-gen
         env:
           AZ_AR_PASSWORD: ${{ secrets.AZ_AR_PASSWORD }}
+      - name: Download dist folders
+        uses: actions/download-artifact@v3
+        if: github.event_name == 'pull_request'
```
Comment (kpanot) [inline .github/workflows/publish.yml:65]: Actually the check is not correct, it should be according to the trigger.
Maybe something like `github.event.release != null` or equivalent to see if you are triggered by `release` event or not

Comment (matthieu-crouzet) [reply .github/workflows/publish.yml:65]: if the event_name is pull_request then it couldn't be a release event 
else event_name would have been release no ?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -36,6 +36,15 @@ jobs:
           AZ_AR_PASSWORD: ${{ secrets.AZ_AR_PASSWORD }}
       - name: Build
         run: yarn build
+      - name: Store dist folders
+        uses: actions/upload-artifact@v3
+        if: github.event.release == null
```
Comment (kpanot) [inline .github/workflows/main.yml:41]: not needed, there is no `release` trigger ;)

**Comment by kpanot** [comment]
Duplicate of #81

**Pull request closed without merging**